### ISSUE TITLE
Enforce file size on Java inputs

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#11163]
From https://github.com/checkstyle/checkstyle/pull/11156#pullrequestreview-845293639

We need to get statistics on how long our Input files and get median value and enforce file size to not above this value.

As we usually do:
Enforce rules with long list of suppression
All suppressions can be resolved in separate PRs

Rationale: big input files a hard to review, hard to debug.

**Comment by Vyom-Yadav** [comment]
@romani Here is some data-
```
Total Num Of Lines: 226357
Total Num Of Files: 1976
Median of number of lines: 44.0
The average number of lines: 114.55313765182186
```
This includes Input files from both `../resources` and `../resources-noncompilable`.
How I generated the data: https://gist.github.com/Vyom-Yadav/76cedd6a5286496e25cf762b31ac2e10

**Comment by romani** [comment]
I am ok to make max limit as 120 lines, I glanced through files, all big files are NOT for good reason.

@pbludov , @rnveach , @nmancus1 , @strkkk , please confirm that you are ok with this.
there definetely might be exceptions, and we can keep them as big files.

**Comment by rnveach** [comment]
I would prefer 200.

Configuration javadoc, package, imports, class definitions, and the necessary blank lines add up.

**Comment by romani** [comment]
We do not use imports that much, header comment is not that big. Please show me input file that is good to be above 120 lines. There might few only.

**Comment by pbludov** [comment]
Long files list: https://gist.github.com/pbludov/fe3786436e177939e2041e3596626296
355 files > 120 lines
200 files > 200 lines

We can start with a limit of 200 lines, and then reduce the limit to 120.

**Comment by romani** [comment]
count of files is not that I worry about. This task will be perfect for introduction to project, so it will be long lasting and done eventually - ok.

Having two iterations: to 200 and then to 120 will be a lot of efforts. Contributor will spend same time to split files from xxxxx to 200 as to xxxx to 120.

**Comment by nmancus1** [comment]
> I am ok to make max limit as 120 lines...

I agree that this is a good limit under normal circumstances. Limiting file lengths will help to make violations more clear, and to reduce/ eliminate extension of existing test inputs.

In PR for this issue, we should decide on big files to keep with comment explaining why for each suppression, then have contributors simply select suppression with no comment to work on.

**Comment by romani** [comment]
First PR should do suppression "until #11163". If we decide to keep big input, we should make special comment.

**Comment by strkkk** [comment]
I think 200 is good enough

**Comment by Vyom-Yadav** [comment]
Here is some more data to help you decide whether it will be 120 or 200.
```
Total Num Of Lines: 54466
The average number of lines: 27.521980798383023
```
This number is composed of the following-
```
/*
SomeCheck
max = (default)200


*/

package com......

public class InputSomeCheck {

    void method() {
    }

}
```
Total num of lines before top-level class declaration - 10 (includes the class declaration)
Total num of empty lines (for readability of code) - 2 (After class declaration)
Total = 10 + 2 = 12
Here is the data for every input file (.java) - https://gist.github.com/Vyom-Yadav/44af5b511fd2d7b89eacacb62fed7ea5

**Comment by romani** [comment]
We come to majority agreement to make it 120.

We can always keep in permanent suppression cases that need more for good reason or we can make limit higher later on.
For now let's try 120.

@Vyom-Yadav , can you send PR that will do validation and placing all files above limit to temporary suppression ?

**Comment by romani** [comment]
@Rahulkhinchi03 , please help with this issue , it is easy, just split Input files into two or few and remove suppression.

**Comment by Rahulkhinchi03** [comment]
@romani Before Starting I need to confirm a few things:

1. I have to limit Input files with 120 lines max + remove suppression for a single check input file and divide the big input files into two to more. 
2. I have to update the UTs for the same.

Any other thing to include in PR?

**Comment by romani** [comment]
Yes, simple update, but would require attention to details and understanding how Check works.

**Comment by Rahulkhinchi03** [comment]
@romani The file `InputJavaDocStyle1 ` has 400 lines of code and It has to be divided into 4 different input files. 

Can you help me in Naming the Files?
PS: We can't do InputJavaDocStyle2, InputJavaDocStyle3 as we have a total of 10 files which have to be distributed into 40 files.

**Comment by romani** [comment]
Please use numbers, I have no better naming model

**Comment by Subzer0570** [comment]
Is this issue completed?

**Comment by rnveach** [comment]
https://github.com/checkstyle/checkstyle/blob/master/config/checkstyle_resources_suppressions.xml#L178
No.

**Comment by Subzer0570** [comment]
Which part on this issue can I work on?
I am new to open source and this is my first issue.

**Comment by romani** [comment]
All files under https://github.com/checkstyle/checkstyle/blob/master/config/checkstyle_resources_suppressions.xml#L178
Should be split in smaller files and removed from suppression

**Comment by Adityasaini10012001** [comment]
I am new to open source , How  I can contribute  to this

**Comment by rnveach** [comment]
@Adityasaini10012001 Look at first post in issue. Read and understand what it is asking. Look over other merged commits in the git history or PR history. Get an idea from the post, how other people completed the issue and you should get an idea of what you need to do. If you have questions you can ask in gitter or github's Checkstyle QnA section.

**Comment by Shubhra0310** [comment]
I want to work on this issue

**Comment by romani** [comment]
@Shubhra0310, if you want to help us to resolve issue, just do comment "I am on it" and send Pull Request.

**Comment by coderdeadpool** [comment]
I am on it.

**Comment by sd9616** [comment]
I am on InputMagicNumber files

**Comment by sd9616** [comment]
I am on InputAtclauseOrder files

**Comment by romani** [comment]
@nitish799 , please choose any Check under suppression and please send PR for single Check.

**Comment by suniti0804** [comment]
List of java input files for which we need to enforce file size -

- [ ] InputAbbreviationAsWordInNameIgnore
- [ ] InputAbbreviationAsWordInNameIgnoreFinal
- [ ] InputAbbreviationAsWordInNameIgnoreFinalKeepStaticFinal
- [ ] InputAbbreviationAsWordInNameIgnoreNonStaticFinal
- [ ] InputAbbreviationAsWordInNameIgnoreStatic
- [ ] InputAbbreviationAsWordInNameIgnoreStaticFinal
- [ ] InputAbbreviationAsWordInNameIgnoreStaticKeepStaticFinal
- [ ] InputAbbreviationAsWordInNameNoIgnore
- [ ] InputAbbreviationAsWordInNameOverridableMethod
- [ ] InputAbbreviationAsWordInNameType
- [ ] InputAbbreviationAsWordInNameType2
- [ ] InputAbbreviationAsWordInNameType3
- [ ] InputAbbreviationAsWordInNameType4
- [ ] InputAbbreviationAsWordInNameType5
- [ ] InputAbbreviationAsWordInNameType6
- [ ] InputAbstractJavadocCorrectParagraph
- [ ] InputAbstractJavadocLeaveToken
- [ ] InputAbstractJavadocNonTightHtmlTags
- [ ] InputAbstractJavadocNonTightHtmlTagsNoViolation
- [ ] InputAbstractJavadocNonTightHtmlTagsVisitCount
- [ ] InputAbstractJavadocPosition
- [ ] InputAbstractJavadocPositionWithSinglelineComments
- [ ] InputAnnotationLocationIncorrect
- [ ] InputAnnotationLocationIncorrect2
- [ ] InputAnnotationLocationIncorrect3
- [ ] InputAnonInnerLength
- [ ] InputAnonInnerLength2
- [ ] InputAntlr4AstRegressionCassandraInputWithComments
- [ ] InputAntlr4AstRegressionCommentsOnAnnotationsAndEnums
- [ ] InputAntlr4AstRegressionFallThroughDefault
- [ ] InputAntlr4AstRegressionKeywordsAndOperators
- [ ] InputAntlr4AstRegressionNewTypeTree
- [ ] InputAntlr4AstRegressionTrickySwitch
- [ ] InputAntlr4AstRegressionTrickySwitchWithComments
- [ ] InputAntlr4AstRegressionUncommon
- [ ] InputAstTreeStringPrinterComments
- [ ] InputAstTreeStringPrinterFullOfBlockComments
- [ ] InputAstTreeStringPrinterJavadocPosition
- [ ] InputAstTreeStringPrinterTextBlocksEscapesAreOneChar
- [ ] InputAvoidEscapedUnicodeCharacters
- [ ] InputAvoidEscapedUnicodeCharacters1
- [ ] InputAvoidEscapedUnicodeCharacters2
- [ ] InputAvoidEscapedUnicodeCharacters3
- [ ] InputAvoidEscapedUnicodeCharacters4
- [ ] InputAvoidEscapedUnicodeCharactersAllEscapedUnicodeCharacters
- [ ] InputCatchParameterNameSimple
- [ ] InputCheckerTabCharacter
- [ ] InputClassFanOutComplexityRemoveIncorrectAnnotationToken
- [ ] InputComments2
- [ ] InputCommentsIndentationCommentIsAtTheEndOfBlock
- [ ] InputCommentsIndentationInMultiblockStructures
- [ ] InputCommentsIndentationInSwitchBlock
- [ ] InputCommentsIndentationSurroundingCode
- [ ] InputCommentsIndentationSurroundingCode2
- [ ] InputCommentsIndentationSurroundingCode3
- [ ] InputConstantNameSimple1
- [ ] InputConstantNameSimple2
- [ ] InputCorrectSummaryJavaDocCheck
- [ ] InputCovariantEquals
- [ ] InputCustomImportOrderDefaultPackage
- [ ] InputCustomImportOrderNoPackage
- [ ] InputCustomImportOrderNoPackage2
- [ ] InputCustomImportOrderNoPackage3
- [ ] InputCustomImportOrderSamePackage
- [ ] InputCustomImportOrderSamePackage2
- [ ] InputCustomImportOrderSamePackageDepth25
- [ ] InputCustomImportOrderSamePackageDepth252
- [ ] InputCustomImportOrderSamePackageDepth253
- [ ] InputCustomImportOrderSamePackageDepth254
- [ ] InputCyclomaticComplexity
- [ ] InputCyclomaticComplexityRecords
- [ ] InputDeclarationOrder
- [ ] InputDeclarationOrderOnlyConstructors
- [ ] InputDeclarationOrderOnlyModifiers
- [ ] InputDefaultComesLast
- [ ] InputDefaultComesLastSkipIfLastAndSharedWithCase
- [ ] InputDesignForExtension
- [ ] InputDesignForExtensionIgnoredAnnotations
- [ ] InputDesignForExtensionOverridableMethods
- [ ] InputEmptyBlockBasic
- [ ] InputEmptyBlockSwitchExpressions
- [ ] InputEmptyCatchBlockDefault
- [ ] InputEmptyCatchBlockDefault2
- [ ] InputEmptyCatchBlockDefaultLF
- [ ] InputEmptyLineSeparator
- [ ] InputEmptyLineSeparator2
- [ ] InputEmptyLineSeparatorNoViolationOnEmptyLineBeforeComments
- [ ] InputEmptyLineSeparatorWithComments
- [ ] InputEmptyLineSeparatorWithComments2
- [ ] InputEqualsAvoidNull
- [ ] InputEqualsAvoidNullIgnoreCase
- [ ] InputEqualsHashCodeSemantic
- [ ] InputFallThrough
- [ ] InputFallThrough3
- [ ] InputFallThroughDefault
- [ ] InputFileLength
- [ ] InputFileLength2
- [ ] InputFileLength3
- [ ] InputFileLength4
- [ ] InputFileTabCharacter
- [ ] InputFileTabCharacterCustomMessage
- [ ] InputFileTabCharacterSimple
- [ ] InputFileTabCharacterSimple1
- [ ] InputFinalClass
- [ ] InputFinalClassClassWithPrivateCtorWithNestedExtendingClassWithoutPackage
- [ ] InputFinalLocalVariableAssignedMultipleTimes
- [ ] InputFinalLocalVariableCheckSwitchExpressions
- [ ] InputFinalLocalVariableFalsePositives
- [ ] InputFinalParameters
- [ ] InputFinalParameters2
- [ ] InputFinalParameters3
- [ ] InputFinalParameters4
- [ ] InputFinalParameters5
- [ ] InputHiddenField1
- [ ] InputHiddenField2
- [ ] InputHiddenField3
- [ ] InputHiddenField4
- [ ] InputHiddenField5
- [ ] InputHiddenField6
- [ ] InputHiddenField7
- [ ] InputHiddenField8
- [ ] InputHiddenFieldLambdas
- [ ] InputHiddenFieldReorder
- [ ] InputIllegalInstantiationBeginTree2
- [ ] InputIllegalInstantiationLang
- [ ] InputIllegalInstantiationNoPackage
- [ ] InputIllegalInstantiationSemantic
- [ ] InputIllegalInstantiationSemantic2
- [ ] InputIllegalTokenText
- [ ] InputImportControlBeginTree2
- [ ] InputIndentationCorrectFieldAndParameter
- [ ] InputIndentationCorrectIfAndParameter
- [ ] InputIndentationCorrectMultipleAnnotationsWithWrappedLines
- [ ] InputIndentationCorrectReturnAndParameter
- [ ] InputIndentationCorrectWhileDoWhileAndParameter
- [ ] InputIndentationCtorCall
- [ ] InputIndentationCustomAnnotation
- [ ] InputIndentationFromGuava
- [ ] InputIndentationFromGuava2
- [ ] InputIndentationInvalidArrayInitIndent
- [ ] InputIndentationInvalidBlockIndent
- [ ] InputIndentationInvalidClassDefIndent
- [ ] InputIndentationInvalidIfIndent
- [ ] InputIndentationInvalidMethodIndent
- [ ] InputIndentationLabels
- [ ] InputIndentationLambda1
- [ ] InputIndentationLongConcatenatedString
- [ ] InputIndentationTryResourcesNotStrict
- [ ] InputIndentationTryWithResourcesStrict
- [ ] InputIndentationValidArrayInitIndent
- [ ] InputIndentationValidBlockIndent
- [ ] InputIndentationValidClassDefIndent
- [ ] InputIndentationValidIfIndent
- [ ] InputIndentationValidMethodIndent
- [ ] InputInnerAssignment
- [ ] InputInnerTypeLastClass
- [ ] InputJava14InstanceofWithPatternMatching
- [ ] InputJava14Records
- [ ] InputJava14SwitchExpression
- [ ] InputJava14TextBlocks
- [ ] InputJava14TextBlocksTabSize
- [ ] InputJavaDocTagContinuationIndentation
- [ ] InputJavaNCSSRecordsAndCompactCtors
- [ ] InputJavaNCSSRecordsMax
- [ ] InputJavaParserFullJavaIdentifierSupport
- [ ] InputJavadocContentLocationTrailingSpace
- [ ] InputJavadocMetadataScraperAnnotationUseStyleCheck
- [ ] InputJavadocMethodExtraThrows
- [ ] InputJavadocMethodIgnoreThrows
- [ ] InputJavadocMethodNoJavadocDefault
- [ ] InputJavadocMethodNoJavadocOnlyPrivateScope
- [ ] InputJavadocMethodNoJavadocProtectedScope
- [ ] InputJavadocMethodProtectedScopeJavadoc
- [ ] InputJavadocMethodPublicOnly
- [ ] InputJavadocMethodPublicOnly1
- [ ] InputJavadocMethodSurroundingAccessModifier
- [ ] InputJavadocMethodTags
- [ ] InputJavadocMethodThrowsDetection
- [ ] InputJavadocMissingWhitespaceAfterAsteriskValidWithTab
- [ ] InputJavadocPackageNoJavadoc
- [ ] InputJavadocParagraphCorrect
- [ ] InputJavadocTagContinuationIndentation
- [ ] InputJavadocTagContinuationIndentationGuavaFalsePositive
- [ ] InputJavadocTypeJavadoc
- [ ] InputJavadocTypeNoJavadoc
- [ ] InputJavadocTypePublicOnly
- [ ] InputJavadocTypePublicOnly1
- [ ] InputJavadocTypeTags
- [ ] InputJavadocVariableNoJavadoc
- [ ] InputJavadocVariableNoJavadoc2
- [ ] InputJavadocVariableNoJavadoc3
- [ ] InputJavadocVariableNoJavadoc4
- [ ] InputJavadocVariableNoJavadoc5
- [ ] InputJavadocVariablePublicOnly
- [ ] InputJavadocVariablePublicOnly2
- [ ] InputJavadocVariableTags
- [ ] InputJavadocWithCrAsNewline
- [ ] InputLeadingAsterisksExtended
- [ ] InputLeftCurlyBraces
- [ ] InputLeftCurlyTestDefault3
- [ ] InputLeftCurlyTestMissingBraces
- [ ] InputLeftCurlyTestNewLine3
- [ ] InputLineLength
- [ ] InputLineLengthLongPackageStatement
- [ ] InputLineLengthSimple
- [ ] InputLineLengthSimple1
- [ ] InputListWithUnclosedItemInUnclosedParagraph
- [ ] InputLocalFinalVariableName
- [ ] InputLocalFinalVariableName1
- [ ] InputLocalVariableName
- [ ] InputMagicNumberAnnotationElement
- [ ] InputMainComplexityOverflow
- [ ] InputMainGenerateXpathSuppressionsTabWidth
- [ ] InputMainSuppressionsStringPrinter
- [ ] InputMemberNameSimple
- [ ] InputMethodCount
- [ ] InputMethodCount1
- [ ] InputMethodLengthCountEmptyIsFalse
- [ ] InputMethodLengthModifier
- [ ] InputMethodLengthSimple
- [ ] InputMethodNameSimple
- [ ] InputMethodParamPad
- [ ] InputMissingJavadocMethodNoJavadoc
- [ ] InputMissingJavadocMethodNoJavadoc2
- [ ] InputMissingJavadocMethodNoJavadoc3
- [ ] InputMissingJavadocMethodTags
- [ ] InputMissingJavadocTypeNoJavadoc1
- [ ] InputMissingJavadocTypeNoJavadoc2
- [ ] InputMissingJavadocTypeNoJavadoc3
- [ ] InputMissingJavadocTypePublicOnly1
- [ ] InputMissingJavadocTypePublicOnly2
- [ ] InputMissingJavadocTypeTags
- [ ] InputMixedCaseOfHtmlTags
- [ ] InputModifierOrder
- [ ] InputModifierOrderIt
- [ ] InputModifierOrderTypeAnnotations
- [ ] InputMultipleStringLiteralsTextBlocks
- [ ] InputNPathComplexity
- [ ] InputNPathComplexityDefault
- [ ] InputNPathComplexityDefault2
- [ ] InputNPathComplexityOverflow
- [ ] InputNeedBraces
- [ ] InputNeedBracesLoopBodyFalse
- [ ] InputNeedBracesLoopBodyTrue
- [ ] InputNeedBracesSingleLineStatements
- [ ] InputNeedBracesTestIt
- [ ] InputNeedBracesTestItWithAllowsOn
- [ ] InputNeedBracesTestSingleLineCaseDefault
- [ ] InputNeedBracesTestSwitchExpression
- [ ] InputNoEnumTrailingComma
- [ ] InputNoFinalizeExtend
- [ ] InputNoFinalizerFallThrough
- [ ] InputNoLineWrapBad
- [ ] InputNoLineWrapBad2
- [ ] InputNoWhitespaceAfterArrayDeclarations2
- [ ] InputNoWhitespaceAfterNewTypeStructure
- [ ] InputNoWhitespaceAfterTestAllTokens
- [ ] InputNoWhitespaceAfterTestAllowLineBreaks
- [ ] InputNoWhitespaceAfterTestDefault
- [ ] InputNoWhitespaceAfterTestTypecast
- [ ] InputNoWhitespaceBeforeDefault
- [ ] InputNoWhitespaceBeforeDot
- [ ] InputNoWhitespaceBeforeDotAllowLineBreaks
- [ ] InputNoWhitespaceBeforeTextBlocksTabIndent
- [ ] InputOneStatementPerLine
- [ ] InputOneStatementPerLineMultiline
- [ ] InputOneStatementPerLineSingleLine
- [ ] InputOperatorWrap
- [ ] InputOuterType
- [ ] InputOuterTypeNumberSimple
- [ ] InputOuterTypeNumberSimple1
- [ ] InputOverloadMethodsDeclarationOrder
- [ ] InputPackageDeclarationNoPackage
- [ ] InputPackageNameSimple
- [ ] InputPackageNameSimple1
- [ ] InputParameterName
- [ ] InputParameterNameOne
- [ ] InputParameterNumberSimple
- [ ] InputParameterNumberSimple2
- [ ] InputParameterNumberSimple3
- [ ] InputParameterNumberSimple4
- [ ] InputParenPad
- [ ] InputParenPadLeftRightAndNoSpace1
- [ ] InputParenPadLeftRightAndNoSpace2
- [ ] InputParenPadLeftRightAndNoSpace3
- [ ] InputParenPadWhitespace
- [ ] InputParenPadWhitespace2
- [ ] InputRecordComponentNumber
- [ ] InputRecordComponentNumberMax1
- [ ] InputRecordComponentNumberMax20
- [ ] InputRecordComponentNumberPrivateModifier
- [ ] InputRedundantImport
- [ ] InputRedundantImportWithoutPackage
- [ ] InputRedundantModifierIt
- [ ] InputRegexpCheckStopEarly
- [ ] InputRegexpMultilineSemantic
- [ ] InputRegexpMultilineSemantic2
- [ ] InputRegexpMultilineSemantic3
- [ ] InputRegexpMultilineSemantic4
- [ ] InputRegexpMultilineSemantic5
- [ ] InputRegexpMultilineSemantic6
- [ ] InputRegexpMultilineSemantic7
- [ ] InputRegexpMultilineSemantic8
- [ ] InputRegexpMultilineSemantic9
- [ ] InputRegexpOnFilenameSemantic
- [ ] InputRegexpSemantic
- [ ] InputRegexpSemantic10
- [ ] InputRegexpSemantic11
- [ ] InputRegexpSemantic12
- [ ] InputRegexpSemantic13
- [ ] InputRegexpSemantic14
- [ ] InputRegexpSemantic2
- [ ] InputRegexpSemantic3
- [ ] InputRegexpSemantic4
- [ ] InputRegexpSemantic5
- [ ] InputRegexpSemantic6
- [ ] InputRegexpSemantic7
- [ ] InputRegexpSemantic8
- [ ] InputRegexpSemantic9
- [ ] InputRegexpSinglelineJavaSemantic
- [ ] InputRegexpSinglelineJavaSemantic2
- [ ] InputRegexpSinglelineJavaSemantic3
- [ ] InputRegexpSinglelineJavaSemantic4
- [ ] InputRegexpSinglelineJavaSemantic5
- [ ] InputRegexpSinglelineJavaSemantic6
- [ ] InputRegexpSinglelineJavaSemantic7
- [ ] InputRegexpSinglelineJavaTrailingComment10
- [ ] InputRegexpSinglelineSemantic
- [ ] InputRegexpSinglelineSemantic2
- [ ] InputRegexpSinglelineSemantic3
- [ ] InputRegexpSinglelineSemantic4
- [ ] InputRegexpSinglelineSemantic5
- [ ] InputRegexpSinglelineSemantic6
- [ ] InputRegexpSinglelineSemantic7
- [ ] InputRegexpSinglelineSemantic8
- [ ] InputRegexpTrailingComment10
- [ ] InputRegressionJavaClass1
- [ ] InputRequireThisEnumInnerClassesAndBugs
- [ ] InputRequireThisEnumInnerClassesAndBugs2
- [ ] InputRequireThisEnumInnerClassesAndBugs3
- [ ] InputRequireThisValidateOnlyOverlappingFalse
- [ ] InputRequireThisValidateOnlyOverlappingTrue
- [ ] InputRightCurlyLeftTestAlone
- [ ] InputRightCurlyLeftTestDefault
- [ ] InputRightCurlyLeftTestNewLine
- [ ] InputRightCurlyLeftTestSame
- [ ] InputRightCurlyLeftTestShouldStartLine2
- [ ] InputRightCurlyOther
- [ ] InputRightCurlyOther2
- [ ] InputRightCurlyTestAloneOrSingleline
- [ ] InputRightCurlyTestWithAnnotations
- [ ] InputSeparatorWrapForTestTrailingWhitespace
- [ ] InputSingleSpaceSeparatorComments
- [ ] InputSingleSpaceSeparatorErrors
- [ ] InputStaticVariableName1
- [ ] InputStaticVariableName2
- [ ] InputSummaryJavadocCorrect
- [ ] InputSummaryJavadocIncorrect
- [ ] InputSummaryJavadocIncorrect2
- [ ] InputSummaryJavadocInlineCorrect
- [ ] InputSummaryJavadocInlineDefault
- [ ] InputSummaryJavadocInlineForbidden
- [ ] InputSuppressionCommentFilter2
- [ ] InputSuppressionCommentFilter3
- [ ] InputSuppressionCommentFilter4
- [ ] InputSuppressionCommentFilter5
- [ ] InputSuppressionCommentFilter6
- [ ] InputSuppressionCommentFilter7
- [ ] InputSuppressionCommentFilter8
- [ ] InputSuppressionCommentFilter9
- [ ] InputTodoCommentSimple
- [ ] InputTypeName
- [ ] InputTypecastParenPadWhitespace
- [ ] InputTypecastParenPadWhitespaceTestSpace
- [ ] InputUnclosedParagraphFollowedByJavadocTag
- [ ] InputUncommentedMainBeginTreePackage2
- [ ] InputUnusedImportsFileInUnnamedPackage
- [ ] InputUnusedImportsSingleWordPackage
- [ ] InputUnusedLocalVariableWithoutPackageStatement
- [ ] InputUpperEllSemantic
- [ ] InputVariableDeclarationUsageDistance
- [ ] InputVariableDeclarationUsageDistanceCheck
- [ ] InputVariableDeclarationUsageDistanceCheckSwitchExpressions
- [ ] InputVariableDeclarationUsageDistanceDefault
- [ ] InputVariableDeclarationUsageDistanceFinal
- [ ] InputVariableDeclarationUsageDistanceGeneral
- [ ] InputVariableDeclarationUsageDistanceRegExp
- [ ] InputVariableDeclarationUsageDistanceScopes
- [ ] InputVisibilityModifierSimple
- [ ] InputWhitespaceAfterDefaultConfig
- [ ] InputWhitespaceAfterTypeCast
- [ ] InputWhitespaceAroundBasic
- [ ] InputWhitespaceAroundBraces
- [ ] InputWhitespaceAroundBraces2
- [ ] InputWhitespaceAroundEmptyTypesAndCycles
- [ ] InputWhitespaceAroundKeywordsAndOperators
- [ ] InputWhitespaceAroundSimple
- [ ] InputXpathFileGeneratorAstFilter
- [ ] InputXpathMapperStringConcat
- [ ] InputXpathQueryGeneratorTabWidth

**Comment by romani** [comment]
@suniti0804 , please reduce list by grouping by Check name. each PR should fix whole Check Inputs.

**Comment by MauryanKansara** [comment]
I'm working on NoEnumTrailingComma.

**Comment by MANISH-K-07** [comment]
I am on abstractJavadocPosition..

**Comment by sktpy** [comment]
I am on InnerAssignment

**Comment by mahfouz72** [comment]
I am on  InputCommentsIndentationCommentIsAtTheEndOfBlock

**Comment by mahfouz72** [comment]
I am on InputCommentIndentationSurroundingCode

**Comment by MANISH-K-07** [comment]
I am on commentsIndentation module remaining files

**Comment by mahfouz72** [comment]
I am on  AnnotationLocation module

**Comment by mahfouz72** [comment]
I am on InputOneStatementPerLineMultiline

**Comment by Vyom-Yadav** [comment]
@romani Agreeing with @nrmancuso's at https://github.com/checkstyle/checkstyle/pull/14369#discussion_r1469097071

> We need to avoid nonsense test method/input file naming conventions like this; if we can't come up with some descriptive name to differentiate these tests, we need to reconsider the content and motivation for them.

We should not allow contributors to just split the file and name them `test...One()` and `test...Two()`.....
Instead, we should split files based on what they logically test. This would make it difficult to separate the test cases, but this would give the contributor a good understanding of what the check does, which is better in the long run.

Please add this to the issue description if you agree.

**Comment by MANISH-K-07** [comment]
@nrmancuso , @Vyom-Yadav 
cc @romani 

I couldn't agree more that the test method names in most of the checks, infact all the checks are nonsense which is decreasing the code readability. It has been mentioned in your discussions changing the test names to something more descriptive of what the test method is actually testing in the files. 
This however is not practically very easy, pls do correct me if I am wrong, because most of the tests test the input files which involve not only the checks that perform the task at hand but also other random violations included which were necessary to comply with the CIs such as pitest. Examples of few such input files are : 
- [InputParameterNumberSimple](https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/parameternumber/InputParameterNumberSimple.java)
- [InputParameterNumberSimple2](https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/parameternumber/InputParameterNumberSimple2.java)
- [InputParameterNumberSimple3](https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/parameternumber/InputParameterNumberSimple3.java)
- [InputParameterNumberSimple4](https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/parameternumber/InputParameterNumberSimple4.java) 

So, it is obvious that we cannot fix the test name based on this.
An alternate to this is **_naming the tests on the basis of the input file name_** which it is testing. As the test is unique to each input file, there would be no duplication. Also, it would be easy to find the specific test according to the name itself instead of searching for the input file path which is currently what is being done. Example of what I am referring to is as follows : 

![image](https://github.com/checkstyle/checkstyle/assets/148562451/efe255b5-7e6d-49bf-89ba-a92a1034d5f3)

**_The name goes as  " test....*InputFileName*.... " which resolves the problem of nonsense test names and also makes the task easy._**

One more thing I would like to mention is, if you think this convention is good, there arise two cases : 
-  test methods which have already been modified as part of Issue #11163 
-  future PRs which are to come as part of the same Issue #11163 

For the second, we could add the naming convention for test methods in the Issue description for new contributors.
As for the first, we could make it a new Issue where the badly named test methods can be modified accordingly.
If you think this is all good, I could create a new Issue concerning this task which would be a good first or second issue as well for the project and new contributors.
Please do correct me if my understanding is wrong anywhere.

**Comment by Vyom-Yadav** [comment]
@MANISH-K-07 Agreed that naming won't be easy as input files are testing multiple things. But the task is to break a big file into logically similar components and name it. `test..InputFile` format is only good if the input file is descriptive as well. `InputCheckOne/Two` is not descriptive.

**Comment by Vyom-Yadav** [comment]
@MANISH-K-07 Sure, we can have an issue for better naming. After all, two difficult things in CS are naming variables and cache invalidation ;)

**Comment by MANISH-K-07** [comment]
> @MANISH-K-07 Sure, we can have an issue for better naming. After all, two difficult things in CS are naming variables and cache invalidation ;)

> Be the change you want to see, anyone is welcome to open issues. Approval is a different story, we will cross that bridge when we encounter it.



@nrmancuso , @Vyom-Yadav 
cc @romani 

I have created a new issue based on our discussions at https://github.com/checkstyle/checkstyle/pull/14369#discussion_r1469873309 .

**_Please review and approve if its all good._**

The existing names are a mess and decrease the code readability !! 
If it could be possible to leave an Open Issue ( similar to https://github.com/checkstyle/checkstyle/issues/13213 or https://github.com/checkstyle/checkstyle/issues/11163 ) regarding this where contributors can update the input file and test names according to the naming convention suggested?.. 

It might be a `good second issue` or so for the next contributors, and at the same time, our "problem at hand" might also get resolved with much lesser effort.
why not let the passionate contributors do the work - the whole point of open source....

**Comment by nrmancuso** [comment]
@MANISH-K-07 @Vyom-Yadav now that we have opened an issue on input file/test method naming conventions, please mark your comments as outdated to clean this issue up.

**Comment by OmarShawky1** [comment]
Is "AbbreviationAsWordInName" check still available to work on?

**Comment by romani** [comment]
There are already numerous PRs merged, doing this, try to figure out by examples.
In short, just cut code by some line before 120 to keep it compiled good, all other should be moved to new Input file.

**Comment by naitikk31** [comment]
Hi, I would like to work on enforcing the file size

**Comment by romani** [comment]
@naitikk31 , please feel welcomed to fix several Inputs

**Comment by ivamly** [comment]
I am on [InputDeclarationOrder.java](https://github.com/checkstyle/checkstyle/blob/8ec5034114cb471ed004efe882e26e57c5891dc7/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/declarationorder/InputDeclarationOrder.java)

**Comment by ivamly** [comment]
I am on DeclarationOrder folder

**Comment by Brijeshthummar02** [comment]
@romani is it still open?

**Comment by Youssefalaa7** [comment]
I am on InputSummaryJavadocInlineDefault

**Comment by aclfe** [comment]
~~Im on InputSummaryJavadocIncorrect2~~ Sent the pr

**Comment by aclfe** [comment]
~~I am on InputSummaryJavadocInlineCorrect~~
Fixing the `summaryjavadoc` files

**Comment by 0xChaitanya** [comment]
If we can't find a way to split, can we split a file as:
Inputxxxx.java : Inputxxxx1.java, Inputxxxx2.java

**Comment by Adam-Gamil** [comment]
I am on InputJavadocVariableTags

**Comment by Adam-Gamil** [comment]
I am on javadocvariable

**Comment by AsthaDhapodkar** [comment]
I am on InputHiddenField2

**Comment by lithops-zty** [comment]
I am on LeftCurly

**Comment by Carbon14-48** [comment]
I am on ConstantName input files.

**Comment by AbdoSol1iman** [comment]
iam on file length input files

**Comment by developer4949-code** [comment]
I am working on this issue. PR #19469 splits `InputLocalFinalVariableName.java` (228 lines) and `InputLocalFinalVariableName1.java` (228 lines) into focused input files under the 120-line limit and removes the corresponding suppressions.

**Comment by developer4949-code** [comment]
Hi @romani, I’ve completed the work for this item in PR #19469 and all checks are green now. Could you please review it when you have time? PR link: https://github.com/checkstyle/checkstyle/pull/19469

**Comment by lithops-zty** [comment]
I'm on NoFinalizerCheck.

**Comment by arturict** [comment]
I am on the WhitespaceAround folder, specifically splitting `InputWhitespaceAroundKeywordsAndOperators.java` and removing its FileLength suppression.

**Comment by arimu1** [comment]
I am on ParameterNumber — PR #20976 splits the oversized `ParameterNumber` inputs under the 120-line limit and drops their `checkstyle_resources_suppressions.xml` entries.

**Comment by NewRecon** [comment]
I'm on `InputAvoidEscapedUnicodeCharacters`

**Comment by AliAkbarBaloch** [comment]
I am on it - checks/blocks/emptycatchblock (InputEmptyCatchBlockDefault.java, InputEmptyCatchBlockDefault2.java, InputEmptyCatchBlockDefaultLF.java).

**Comment by pascofire** [comment]
I'm on InputJavadocTypeTags

**Comment by SidheshwarSarangal** [comment]
I am on it — taking the RegexpOnFilenameCheck input InputRegexpOnFilenameSemantic.java and its corresponding FileLength suppression. I released the initially selected RegexpCheck input because it is an intentional PIT mutation-test fixture.
